src/ngraph_builder.cc

```diff
@@ -4326,6 +4307,38 @@ Status Builder::TranslateGraph(
   //
   ng_function = make_shared<ng::Function>(ng_result_list, ng_parameter_list);
 
+  // Get the serialized ops and stored the allreduce ops to a vector and
+  ng::NodeVector allreduce_op_list;
+  for (const shared_ptr<ng::Node>& node : ng_function->get_ordered_ops()) {
+    if (node->description() == "AllReduce") {
+      allreduce_op_list.push_back(node);
+// Diagnosis to check the input, dependencies, and output for each
+// Allreduce op
+#if defined NGRAPH_DISTRIBUTED
```

Comment:
Please move the #if defined NGRAPH_DISTRIBUTED to line 4309 i.e. the for loop to get the all_reduce node etc can all be moved inside the check?